### DISCUSSION TITLE
How to test service_data output

### DISCUSSION THREAD

**Discussion by mihatsu** [ExperienceLovelace/ha-floorplan#345] [General]
Hi,

I tried to search answer because this must basic thing but I could not find anything. I have many times problems to write service_data line and I don't know how to debug it. How I can call that I see what is the output?

For example how do I see what this service_data output is?

- entities:
        - sensor.garage_temperature_temperature
      state_action:
        - service: floorplan.text_set
          service_data: >-
            ${(entity.state !== undefined) ? Math.round(entity.state * 10) / 10
            + "°" : "unknown"}

**Comment by OzGav** [comment]
It is JavaScript. You could use this to check for syntax errors? (Just remove the $ and curly braces) https://esprima.org/demo/validate.html

**Comment by exetico** [comment]
Hello. You'er able to console.log this, and return the value itself instead. If you look through my videos (find the links in the tutorial category), you should find a example.

**Reply by exetico** [reply]
Take a look at:
https://github.com/exetico/Home-Assistant_Attachments-for-YouTube/blob/main/6_Format-time_date.time-entity-from-military-to-AM_PM-in-state_action/case3-final.yaml

If the last part was saved to a variable first, you could console log it before return it:

```yaml
state_action:
            - service: floorplan.text_set
              service_data:
                text: |
                  > 
                  console.log(entity.state);
                  const dateTimeString = "1970-01-01 " + entity.state // You'll need to provide a date string to new Date()
                  const dateTimeParsed = new Date(dateTimeString); // New date object created here
                  return util.date.strftime('%I:%M %p', dateTimeParsed);
```

I'm on my phone, so its hard to adjust it properly.

```yaml
state_action:
            - service: floorplan.text_set
              service_data:
                text: |
                  > 
                  console.log(entity.state);
                  const dateTimeString = "1970-01-01 " + entity.state // You'll need to provide a date string to new Date()
                  const dateTimeParsed = new Date(dateTimeString); // New date object created here
                  const returnValueDebug = util.date.strftime('%I:%M %p', dateTimeParsed);
                  return returnValueDebug;
```

Now you can open up your inspect tool in your browser, and see the value in the console, every time the code executes.

Also check 18:58 here, if you've not used the Inspect tool before:
https://youtu.be/5ue9J4WG2a8

And 2:30 here:
https://youtu.be/GHcgwW-OOo8

**Reply by mihatsu** [reply]
got it work, thx! nice tutorials!
